src/utils/DateUtils/DateFormatter.js

```diff
@@ -0,0 +1,114 @@
+// @flow strict
+
+import { NativeModules, Platform } from 'react-native';
+import DateUtils from './DateUtils';
+
+import 'intl'; // Polyfill because of Android
+import 'intl/locale-data/complete';
+
+type FormatterConfig = {|
+  +weekday?: 'narrow' | 'short' | 'long',
+  +year?: 'numeric' | '2-digit',
+  +month?: 'numeric' | '2-digit' | 'narrow' | 'short' | 'long',
+  +day?: 'numeric' | '2-digit',
+  +hour?: 'numeric' | '2-digit',
+  +minute?: 'numeric' | '2-digit',
+  +second?: 'numeric' | '2-digit',
+|};
+
+let DEVICE_LOCALE;
+
+switch (Platform.OS) {
+  case 'ios':
+    DEVICE_LOCALE = NativeModules.SettingsManager.settings.AppleLocale.replace(
+      '_',
+      '-'
+    );
+    break;
+
+  case 'android':
+    DEVICE_LOCALE = NativeModules.I18nManager.localeIdentifier.replace(
+      '_',
+      '-'
+    );
+    break;
+  default:
+    DEVICE_LOCALE;
+}
+
+function regularDate(date: Date) {
+  return Intl.DateTimeFormat(DEVICE_LOCALE, {
+    timeZone: 'UTC', // this is very important!
+    weekday: 'short',
+    month: 'numeric',
+    day: 'numeric',
+  }).format(date);
+}
+
+function shortDate(date: Date) {
+  return Intl.DateTimeFormat(DEVICE_LOCALE, {
+    timeZone: 'UTC', // this is very important!
+    month: 'numeric',
+    day: 'numeric',
+  }).format(date);
+}
+
+function time(date: Date) {
+  return Intl.DateTimeFormat(DEVICE_LOCALE, {
+    timeZone: 'UTC', // this is very important!
+    hour: 'numeric',
+    minute: 'numeric',
+  }).format(date);
+}
+
+function birthday(date: Date) {
+  return Intl.DateTimeFormat(DEVICE_LOCALE, {
+    timeZone: 'UTC', // this is very important!
+    year: 'numeric',
+    month: 'numeric',
+    day: 'numeric',
+  }).format(date);
+}
+
+function custom(date: Date, config: FormatterConfig) {
+  return Intl.DateTimeFormat(DEVICE_LOCALE, {
+    ...config,
+    timeZone: 'UTC', // this is very important!
+  }).format(date);
+}
```

Comment:
What about something like?
```
createDateFormater = config => date => {
  return function dateFromtater(date) {
    return Intl.DateTimeFormat(DEVICE_LOCALE, {
      ...config,
      timeZone: "UTC" // this is very important!
    }).format(date);
  };
};

const birthday = createDateFormater({
  year: "numeric",
  month: "numeric",
  day: "numeric"
});

```

Reply:
Isn't the Intl somehow patched in our mobile repo? Because of some troubles with android? cc @ferrannp

Reply:
It's patched there as well, on iOS works fine. Will have to check Android.

Reply:
https://github.com/kiwicom/mobile/blob/master/scripts/configureApplication.js#L87

Reply:
https://github.com/kiwicom/universal-components/blob/e768c49fb9d0ebffe4b82c966c3eaf531aad1743/scripts/postInstall.js

Reply:
I would be happy to have @FilipMessa's proposal to make the code more succinct. At the end of the day, it's achieving the same.

Reply:
Ok, I can rewrite it, it's not an issue. I was just thinking we would like to have the same implementation in two repos. :)